src/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/DiagnosticSourceEventSource.cs

```diff
@@ -640,8 +652,7 @@ private void Dispose()
                             foreach (var property in curTypeInfo.DeclaredProperties)
                             {
                                 var propertyType = property.PropertyType;
-                                if (propertyType == typeof(string) || propertyType.GetTypeInfo().IsPrimitive)
-                                    newSerializableArgs = new TransformSpec(property.Name, 0, property.Name.Length, newSerializableArgs);
+                                newSerializableArgs = new TransformSpec(property.Name, 0, property.Name.Length, newSerializableArgs);
```

Comment:
Looks like some tests need to be updated to account for this change.  Other than that and the typos and whatnot, LGTM.